### ISSUE TITLE
Can't reproduce sd3 OCR results

### ISSUE THREAD

**Issue by kfirgoldberg** [yifan123/flow_grpo#227]
Hi, 
I'm trying to run the base setup provided for sd3 with OCR reward on 8xH200:
```
    config.pretrained.model = init_ckpt_dir
    config.sample.num_steps = 10
    config.sample.eval_num_steps = 40
    config.sample.guidance_scale = 4.5

    config.resolution = 512
    config.sample.train_batch_size = 9
    config.sample.num_image_per_prompt = 24
    config.sample.num_batches_per_epoch = int(
        48 / (gpu_number * config.sample.train_batch_size / config.sample.num_image_per_prompt)
    )
    assert config.sample.num_batches_per_epoch % 2 == 0, (
        "Please set config.sample.num_batches_per_epoch to an even number! This ensures that config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch / 2, so that gradients are updated twice per epoch."
    )
    config.sample.test_batch_size = 16  # 16 is a special design, the test set has a total of 1018, to make 8*16*n as close as possible to 1018, because when the number of samples cannot be divided evenly by the number of cards, multi-card will fill the last batch to ensure each card has the same number of samples, affecting gradient synchronization.

    config.train.batch_size = config.sample.train_batch_size
    config.train.gradient_accumulation_steps = config.sample.num_batches_per_epoch // 2
    config.train.num_inner_epochs = 1
    config.train.timestep_fraction = 0.99
    # kl loss
    config.train.beta = 0.04
    # Whether to use the std of all samples or the current group's.
    config.sample.global_std = True
    # Whether to use the same noise for the same prompt
    config.sample.same_latent = False
    config.train.ema = True
    # A large num_epochs is intentionally set here. Training will be manually stopped once sufficient
    config.save_freq = 60  # epoch
    config.eval_freq = 60
    # config.save_dir = 'logs/ocr/sd3.5-M'
    config.save_dir = "/opt/ml/checkpoints"
    config.reward_fn = {
        "ocr": 1.0,
    }

    config.prompt_fn = "general_ocr"

    config.per_prompt_stat_tracking = True
```

But the reward is pretty much stuck in place, both during training and eval:

<img width="2528" height="1328" alt="Image" src="https://github.com/user-attachments/assets/81307d50-7955-423c-867c-25997d218c99" />
<img width="2528" height="1328" alt="Image" src="https://github.com/user-attachments/assets/1a9e42b9-4f63-4e3f-be00-f95fcf06eb5f" />

I can provide other metrics from the Wandb if it helps.

Additional details:
I launched the run the following way
`torchrun --nnodes 1 --nproc_per_node 8 scripts/train_sd3.py --config config/grpo.py:general_ocr_sd3`
and manually set the accelerator mixed precision to `fp16`.

Can you please help me figure out what's wrong here?
and manually set the
